Add version bounds

rkrzr opened the pull request:
This PR adds version bounds, so that we can upload the package to Hackage.

The bounds were generated with `cabal gen-bounds` and then manually widened.

fatho commented on alfred-margaret.cabal:
```diff
@@ -47,18 +47,20 @@ library
-                     , unordered-containers
-                     , vector
+  build-depends:
+      base             >= 4.7 && < 5
+    , containers       >= 0.6.2 && < 0.7
+    , deepseq          >= 1.4.4 && < 1.5
+    , hashable         >= 1.3.0 && < 1.4
+    , primitive        >= 0.7.0 && < 0.8
+    , text             >= 1.2.4 && < 1.3
+    , unordered-containers  >= 0.2.10 && < 0.3
+    , vector           >= 0.12.1 && < 0.13
   ghc-options:         -Wall -Wincomplete-record-updates -Wincomplete-uni-patterns -O2
   default-language:    Haskell2010
   if flag(aeson) {
-  build-depends:       aeson
+  build-depends:
+      aeson            >= 1.4.7 && < 1.6
```
This version bounds are not compatible with the the stage LTS 13.10 of which we say above that we are compatible. I relaxed them a bit more, these work on that older LTS as well:

```
  build-depends:
      base             >= 4.7 && < 5
    , containers       >= 0.6 && < 0.7
    , deepseq          >= 1.4 && < 1.5
    , hashable         >= 1.2.7 && < 1.4
    , primitive        >= 0.6.4 && < 0.8
    , text             >= 1.2.3 && < 1.3
    , unordered-containers  >= 0.2.9 && < 0.3
    , vector           >= 0.12 && < 0.13
  ghc-options:         -Wall -Wincomplete-record-updates -Wincomplete-uni-patterns -O2
  default-language:    Haskell2010
  if flag(aeson) {
  build-depends:
      aeson            >= 1.4.2 && < 1.6
  cpp-options:         -DHAS_AESON
  }
```

fatho reviewed (changes requested):
Thanks! I suggested a bit more relaxed version bounds that also work with LTS 13.10 below:

Also, I noticed that the message of commit https://github.com/channable/alfred-margaret/pull/9/commits/fe2903d60574e41078469877248958c9b87fd7d3
>  Upgrade from lts-13.10 to lts-16.18 

is no longer accurate (after rebasing presumably), `master` is already on 16.18.

rkrzr commented:
> is no longer accurate (after rebasing presumably), master is already on 16.18.

True, I changed the commit message now. I also squashed the two commits to add version bounds and added your bounds.

fatho reviewed (approved):
LGTM
